helix-core/src/position.rs

```diff
@@ -169,6 +168,12 @@ pub fn pos_at_visual_coords(text: RopeSlice, coords: Position, tab_width: usize)
     line_start + col_char_offset
 }
 
+fn rope_graphemes(text: RopeSlice, pos: usize, row: usize) -> RopeGraphemes {
+    let row_start = text.line_to_char(row);
+    let pos = ensure_grapheme_boundary_prev(text, pos);
+    RopeGraphemes::new(text.slice(row_start..pos))
+}
```

Comment:
This is not so much code that it needs a helper function IMO, and it's a bit confusing that there are other calls to RopeGraphemes::new that don't use rope_graphemes in this module. This can be inlined into the two callsites

Reply:
Respectfully, I disagree. This function encapsulates an important concept that `text` should be sliced taking into account grapheme boundaries. Maybe the name for the function and the place should be different but otherwise...

I don't know why in other places that's not the case, haven't looked deep.

Let's close this PR because neither I nor you don't have time to argue.